code/ui/Hud.cs

```diff
@@ -51,55 +48,60 @@ private void OnPlayerSpawned(TTTPlayer player)
                 return;
             }
 
-            Current.AliveHudPanel.Enabled = true;
+            AliveHudPanel.Enabled = true;
+            GeneralHudPanel.GetCrosshair().Enabled = true;
         }
 
         [Event("tttreborn.player.died")]
-        private void OnPlayerDied(TTTPlayer player)
+        private void OnPlayerDied(TTTPlayer deadPlayer)
         {
-            if (player != Local.Client.Pawn)
+            if (deadPlayer != Local.Client.Pawn)
             {
                 return;
             }
 
-            Current.AliveHudPanel.Enabled = false;
+            AliveHudPanel.Enabled = false;
+            GeneralHudPanel.GetCrosshair().Enabled = false;
         }
 
         public class GeneralHud : Panel
         {
-            public GeneralHud(Sandbox.UI.Panel parent) : base(parent)
+            public GeneralHud() : base()
             {
-                Parent = parent;
-
-                Parent.AddChild<RadarDisplay>();
-                Parent.AddChild<Crosshair>();
-                Parent.AddChild<PlayerRoleDisplay>();
-                Parent.AddChild<PlayerInfoDisplay>();
-                Parent.AddChild<InventoryWrapper>();
-                Parent.AddChild<ChatBox>();
-
-                Parent.AddChild<VoiceChatDisplay>();
-                Parent.AddChild<GameTimerDisplay>();
-
-                Parent.AddChild<VoiceList>();
+                AddClass("fullscreen");
+                AddChild<Crosshair>();
+                AddChild<RadarDisplay>();
+                AddChild<PlayerRoleDisplay>();
+                AddChild<PlayerInfoDisplay>();
+                AddChild<InventoryWrapper>();
+                AddChild<ChatBox>();
+
+                AddChild<VoiceChatDisplay>();
+                AddChild<GameTimerDisplay>();
+
+                AddChild<VoiceList>();
+
+                AddChild<InfoFeed>();
+                AddChild<InspectMenu>();
+                AddChild<PostRoundMenu>();
+                AddChild<Scoreboard>();
+                AddChild<Menu.Menu>();
+            }
 
-                Parent.AddChild<InfoFeed>();
-                Parent.AddChild<InspectMenu>();
-                Parent.AddChild<PostRoundMenu>();
-                Parent.AddChild<Scoreboard>();
-                Parent.AddChild<Menu.Menu>();
+            public Panel GetCrosshair()
+            {
+                return (Panel) GetChild(0);
```

Comment:
dk whether Crosshair should be the first element nor handled in this way. E.g. the radar points should be before the crosshair.

Reply:
I am not a fan of how I had to do it either, I just wish that I could just turn on and off the parent, but it's giving me hell when I do it.